index.js

```diff
@@ -2585,12 +2605,26 @@
   //. > S.mult(4, 2)
   //. 8
   //. ```
-  S.mult =
+  var mult = S.mult =
   def('mult',
       {},
       [$.FiniteNumber, $.FiniteNumber, $.FiniteNumber],
       function(a, b) { return a * b; });
 
+  //# product :: Foldable f => f FiniteNumber -> FiniteNumber
+  //.
+  //. Returns the product of a list (finite) numbers.
+  //.
+  //. ```javascript
+  //. > S.product([1, 2, 3, 4, 5])
+  //. 120
+  //. ```
+  S.product =
+  def('product',
+      {},
+      [$.Array($.FiniteNumber), $.FiniteNumber],
```

Comment:
Interesting, I found a bug in the `Foldable` type check function. I think we never saw it before because it always short-circuited on the array check.